File: simuration/class_7_10/Gauss-Jordan_Elimination.py
import numpy as np

def Gauss_Jordan_Elimination(X):
    # declare dimension
    N = 3

    # k repeats to row direction
    for k in range(N):

        # j repeats to column direction
        for j in range(k + 1, N + 1):
            X[k][j] = X[k][j] / X[k][k]

            # i repeats to row direction
            # become 0 except [k][k]element
            for i in range(N):
                if i != k:
                    X[i][j] = X[i][j] - X[k][j] * X[i][k]
    return X

if __name__ == '__main__':

    #A = np.array([[2, 3, 4], [3, 5, 2], [4, 3, 30]], dtype=float)
    #B = np.array([[6], [5], [32]], dtype=float)
    A = np.array([[7, 1, 2], [1, 8, 3], [2, 3, 9]], dtype=float)
    B = np.array([[10], [8], [6]], dtype=float)

    # prepare matrix for calculation
    X = np.hstack([A, B])
    print(X)

    X = Gauss_Jordan_Elimination(X)

    print(f'x1 = {X[0][3]}, x2 = {X[1][3]}, x3 = {X[2][3]}')
    print(X)

File: simuration/class_7_10/Gauss-Jordan_Elimination.c
#include <stdio.h>

void Gauss_Jordan_Elimination(double A[3][4]);

int main(void)
{
	double A[3][4] = {{2, 3, 4, 6}, {3, 5, 2, 5}, {4, 3, 30, 32}};
	
	for (int i = 0; i < 3; i++) {
		for (int j = 0; j < 4; j++) {
			printf("%4.f", A[i][j]);
		}
		printf("\n");
	}

	Gauss_Jordan_Elimination(A);

	for (int i = 0; i < 3; i++) {
		for (int j = 0; j < 4; j++) {
			printf("%4.f", A[i][j]);
		}
		printf("\n");
	}
	
	return 0;
}

void Gauss_Jordan_Elimination(double A[3][4])
{
	int N = 3;

	for (int k = 0; k < N; k++) {
		printf("k = %d\n", k);
		for (int j = k + 1; j < N + 1; j++) {
			printf("j = %d\n", j);
			A[k][j] = A[k][j] / A[k][k];
			for (int i = 0; i < N; i++) {
				printf("i = %d\n", i);
				if (i != k) {
					A[i][j] = A[i][j] - A[k][j] * A[i][k];
				}
			}
		}
	}
}
